### PR TITLE
Add support for histogram and scalar binop

### PR DESCRIPTION
**Pull request by fpetkovski**
This commit adds support for operations between a histogram and a scalar.

### CODE REVIEW

File: execution/binary/scalar.go [hunk 1]
```diff
@@ -138,6 +140,19 @@ func (o *scalarOperator) Next(ctx context.Context) ([]model.StepVector, error) {
 			}
 			step.AppendSample(o.pool, vector.SampleIDs[i], val)
 		}
+
+		for i := range vector.HistogramIDs {
+			scalarVal := math.NaN()
```
Comment (MichaHoffmann) [inline execution/binary/scalar.go:145]: couldnt we move `scalarVal` out of the `for i := range` loop since its not dependent on i and used for samples and histograms?

Comment (fpetkovski) [reply execution/binary/scalar.go]: Good point, updated 👍

File: execution/binary/table.go [hunk 2]
```diff
@@ -201,6 +199,42 @@ func newOperation(expr parser.ItemType, vectorBinOp bool) (operation, error) {
...
+	var operation histogramFloatOperation
+	if scalarSide == ScalarSideLeft {
+		operation = rhsHistogramOperations[t]
+	} else {
+		operation = lhsHistogramOperations[t]
+	}
+	if operation != nil {
+		return operation
+	}
+	return undefinedHistogramOp
```
Comment (MichaHoffmann) [inline execution/binary/table.go:235]: shouldnt that return something like "unimplemented" to trigger promql fallback?

Comment (fpetkovski) [reply execution/binary/table.go:235]: The fallback can only be applied at plan compilation time when we detect that a function or operator is implemented in the engine. In this specific case, we want to cover operators that are simply not "defined" between a histogram and float operand, even in Prometheus. 

For example, something like a % b is only defined when both a and b are floats. But to know their type, we have to actually execute the query and decode chunks. To further complicate matters, a series can have interleaved histograms and floats, so for some samples the operation will be defined and for others it won't. 

So bottom line, it should be fine to produce a nil value when an operation between a histogram and float is simply not defined, as in the a % b example.

Comment (MichaHoffmann) [reply execution/binary/table.go:235]: ah! ok i understand; can you add a small test to ensure that we are behaving like prometheus in that case?

Comment (MichaHoffmann) [reply execution/binary/table.go:235]: nvm, i have found 
```
If either operator has to sum up a mix of histogram samples and float samples, the corresponding vector element is removed from the output vector entirely.
```

Comment (MichaHoffmann) [reply execution/binary/table.go:235]: ok, i think this will just work!

**Review by MichaHoffmann — APPROVED** [review]
lgtm

**Review by saswatamcode — APPROVED** [review]
LGTM!
